docs/CALENDAR-FORMAT.md

```diff
@@ -0,0 +1,479 @@
+  errors: string[];
+  warnings: string[];
+}
+```
+
+### Common Validation Errors
+
+- Missing required fields (`id`, `translations`, `year`, `months`, `weekdays`)
+- Invalid month lengths (must be positive integers)
+- Invalid weekday count (must have at least 1 weekday)
+- Duplicate month or weekday names
+- Invalid leap year configuration
+- Circular variant references
+
+### Error Handling
+
+```javascript
+try {
+  const calendar = await game.seasonsStars.api.loadExternalCalendar(externalId);
+} catch (error) {
+  if (error.name === 'ValidationError') {
+    console.error('Calendar validation failed:', error.message);
+  } else if (error.name === 'NetworkError') {
+    console.error('Failed to load calendar:', error.message);
+  } else if (error.name === 'TimeoutError') {
+    console.error('Calendar loading timeout:', error.message);
+  }
+}
+```
+
+### Cache Management
+
+External calendars are cached with configurable duration:
+
+```javascript
+// Default cache settings
+{
+  defaultCacheDuration: 7 * 24 * 60 * 60 * 1000, // 1 week
+  maxCacheSize: 100,
+  requestTimeout: 30000,
+  autoUpdate: false,
+  updateInterval: 24 * 60 * 60 * 1000 // 24 hours
+}
+```
+
+#### Development Mode Cache Skipping
+
+Cache is automatically skipped when:
+
+- `game.data.world.flags['seasons-and-stars']?.devMode` is true
+- URL contains 'foundrytest'
+- Module protocol with `skipModuleCache: true`
+
+## Best Practices
+
+### Calendar Design
+
+1. **Unique IDs**: Use descriptive, unique identifiers
+2. **Complete Translations**: Provide comprehensive localization
+3. **Logical Structure**: Design months and weekdays logically
+4. **Validation**: Test calendars thoroughly before distribution
+
+### External Calendar Hosting
+
+1. **HTTPS Only**: Always use HTTPS for security
+2. **CORS Headers**: Configure proper CORS headers for web hosting
+3. **Stable URLs**: Use stable, long-term URLs for calendar hosting
+4. **Version Control**: Use GitHub or similar for version control
+
+### Performance Optimization
+
+1. **Cache Strategy**: Use appropriate cache durations
+2. **Compression**: Compress JSON files for faster loading
+3. **CDN**: Use CDN for better global performance
+4. **Index Organization**: Organize large collections with clear indexes
+
+### Security Considerations
+
+1. **Trusted Sources**: Only load calendars from trusted sources
+2. **Input Validation**: Always validate external calendar data
+3. **Error Handling**: Implement robust error handling
+4. **Rate Limiting**: Respect API rate limits for external services
+
+### Module Integration
+
+```javascript
+// Register external calendar sources in your module
+Hooks.on('seasons-stars:ready', async () => {
+  if (game.seasonsStars.manager?.externalRegistry) {
+    await game.seasonsStars.manager.externalRegistry.addExternalSource({
+      protocol: 'module',
+      location: 'my-module/calendars/special.json',
+      label: 'My Module Calendar',
+      trusted: true,
+    });
+  }
+});
+```
```

Comment:
If we're doing it in a hook, S&S could pass in a function to call to register a calendar or calendars/calendar index.

```javascript
Hooks.on('season-stars:loadCalendars', ({ addCalendar }) => addCalendar( ... ));
```